src/Groups/schur_index.jl

```diff
@@ -111,6 +115,79 @@ function _Riese_Schmid_type(G::GAPGroup)
 end
 
 
+# Return the 2-local Schur index of `chi` via a
+# reduction from (G, chi) = (G_0, chi_0) to (G_n, chi_n)
+# such that chi_n is faithful and all proper subgroups H of G_n satisfy
+# that for all eta in Irr(H),
+# scalar_product(eta^{G_n}, chi_n) [\Q_2(chi, eta):\Q_2(chi)] is even.
+# - If chi_i is not faithful then G_{i+1} = G_i/ker(chi_i)
+#   and chi_{i+1} in Irr(G_{i+1}) inflates to chi_i.
+#   In this case, chi_i and chi
+# - If chi_i is faithful then G_{i+1} is a subgroup of G_i such that
+#   scalar_product(chi_{i+1}^{G_i}, chi_i) [\Q_2(chi_i, chi_{i+1}):\Q_2(chi_i)]
+#   is odd.
+#   In this case, if [\Q_2(chi_i, chi_{i+1}):\Q_2(chi_{i+1})] is even then
+#   return m_2(chi) = 1.
+#   Otherwise m_2(chi) = m_2(chi_n), which is computed by Theorem 5.2 in
+#   [Ung19].
+#
+function _Riese_Schmid_reduction(chi::GAPGroupClassFunction)
+  minimal = false
+  tblG = parent(chi)
+  G = group(tblG)
+  while !minimal
+    K = class_positions_of_kernel(chi)
+    if length(K) > 1
+      F, fus = quo(tblG, K)
+      chi = class_function(F, values(chi)[_projection_of_class_fusion(fus)])
+      tblG = parent(chi)
+      G = group(tblG)
+    else
```

Comment:
Would it make sense to rewrite this like this, to reduce the nesting depth and "improve code locality" (i.e. make it easier to understand the code)
```suggestion
    while length(K) > 1
      F, fus = quo(tblG, K)
      chi = class_function(F, values(chi)[_projection_of_class_fusion(fus)])
      tblG = parent(chi)
      G = group(tblG)
      K = class_positions_of_kernel(chi)
    end
```